### PR TITLE
feat(entity): allow custom id property names. 

### LINKED ISSUE #142: allow other properties than 'id' as id on entities

### PR DESCRIPTION
**Pull request by doktordirk**
Adds @idProperty() decorator to set a custom id property name.

closes https://github.com/SpoonX/aurelia-orm/issues/142

### CODE REVIEW

File: src/entity.js [hunk 1]
```diff
@@ -471,7 +514,7 @@ function asObject(entity, shallow) {
       }
 
       if (value.id) {
-        pojo[propertyName] = value.id;
+        pojo[propertyName] = value.id; // weird. entities always have an id!?
```
Comment (doktordirk) [inline src/entity.js]: while all tests works, i don't know what that is about

File: src/entity.js [hunk 2]
```diff
@@ -570,7 +613,7 @@ function getCollectionsCompact(forEntity, includeNew) {
       }
 
       if (entity.id) {
-        collections[index].push(entity.id);
+        collections[index].push(entity.id);  // again weird. see above
```
Comment (doktordirk) [inline src/entity.js]: same here

File: src/entity.js (hunk 1 above)
Comment (doktordirk) [inline src/entity.js]: ah, i see that's with new collection entities which don't have an id yet. hmm

File: src/entity.js [hunk 3]
```diff
@@ -569,10 +614,12 @@ function getCollectionsCompact(forEntity, includeNew) {
         return;
       }
 
-      if (entity.id) {
-        collections[index].push(entity.id);
-      } else if (includeNew && entity instanceof Entity) {
-        collections[index].push(entity);
+      if (entity instanceof Entity) {
```
Comment (RWOverdijk) [inline src/entity.js]:
``` js
if (!entity instanceof Entity) { 
  return;
}
```

File: src/entity.js [hunk 4]
```diff
@@ -470,10 +513,12 @@ function asObject(entity, shallow) {
         return;
       }
 
-      if (value.id) {
-        pojo[propertyName] = value.id;
-      } else if (value instanceof Entity) {
-        pojo[propertyName] = value.asObject();
+      if (value instanceof Entity) {
```
Comment (RWOverdijk) [inline src/entity.js:522]: && value.getId()

No need for extra nesting, it's a simple conditional.

File: doc/api_entity.md [hunk 5]
```diff
@@ -266,7 +266,7 @@ The resource it belongs to (string).
 
 Get the name of this entity.
 
-**Note:** Also works when called statically on custom entities.
+**Note:** Also works when called statically on an Entity class.
```
Comment (RWOverdijk) [inline doc/api_entity.md:269]: > This works when called statically on any instance of `Entity`.

File: doc/api_entity.md [hunk 6]
```diff
@@ -250,7 +250,7 @@ Itself to allow chaining.
 
 Get the resource this entity belongs to.
 
-**Note:** Also works when called statically on custom entities.
+**Note:** Also works when called statically on an Entity class.
```
Comment (RWOverdijk) [inline doc/api_entity.md:253]: > This works when called statically on any instance of `Entity`.

**Comment by RWOverdijk** [comment]
Minor feedback. After than 👍

**Pull request merged**